extensions/moderation/in_voice_moderation.py

```diff
@@ -0,0 +1,203 @@
+        self.deafed = self.user.voice.deaf
+        if self.deafed:
+            self.emoji = ":hear_no_evil:"
+            self.style = discord.ButtonStyle.red
+        else:
+            self.emoji = ":monkey_face:"
+            self.style = discord.ButtonStyle.green
+
+    async def callback(self, interaction):
+        if self.user.voice.channel == self.channel_ctx:
+            if self.deafed:
+                await self.undeaf()
+                self.emoji = ":monkey_face:"
+                self.style = discord.ButtonStyle.green
+            else:
+                await self.deaf()
+                self.emoji = ":hear_no_evil:"
+                self.style = discord.ButtonStyle.red
+        await interaction.response.edit_message(view=self.view)
+
+    async def deaf(self):
+        await self.user.edit(deafen=True)
+        self.deafed = True
+
+    async def undeaf(self):
+        await self.user.edit(deafen=False)
+        self.deafed = False
+
+
+class MuteView(discord.ui.View):
+    def __init__(self, channel_ctx: discord.VoiceChannel):
+        super().__init__()
+        self.channel = channel_ctx
+        self.user_in_channel = self.channel.members
+        container = discord.ui.Container()
+        for user in self.user_in_channel:
+            if user.bot:
+                continue
+            mute = discord.ui.Section(
+                discord.ui.TextDisplay(content=user.display_name), accessory=Mute(user, ctx=channel_ctx)
+            )
+            deaf = discord.ui.Section(
+                discord.ui.TextDisplay(content=user.display_name), accessory=Deaf(user, ctx=channel_ctx)
+            )
+            container.add_item(item=mute)
+            container.add_item(item=deaf)
+            container.add_separator()
+        self.add_item(container)
+
+
+class DnDManager(commands.Cog):
+    def __init__(self, client):
+        self.client: Bot = client
+        self.logger = CustomLogger(self.qualified_name, self.client.boot_time)
+        # {
+        #   channel_id: user
+        # }
+        self.claimed = {}
+        self.requested_message = {}
+
+    @commands.slash_command(name="claim", description="Claim this channel to moderate it")
+    async def claim(self, ctx: discord.ApplicationContext):
+        if isinstance(ctx.channel, discord.VoiceChannel):
+            try:
+                guild_claim = self.claimed[ctx.channel_id]
+                await ctx.response.send_message(
+                    f"This channel is already claimed by {guild_claim.display_name}", ephemeral=True, delete_after=5.0
+                )
+            except KeyError:
+                self.claimed[ctx.channel_id] = ctx.user
+                await ctx.response.send_message("You claimed this channel", ephemeral=True, delete_after=5.0)
+        else:
+            await ctx.response.send_message("This is not a voice channel", ephemeral=True, delete_after=5.0)
+
+    @commands.slash_command(name="unclaim", description="Unclaims the channel")
```

Comment:
should we move it to a subcommand like `/mod voice moderate` ?!
maybe enhance description